dev/fattest.simplicity/src/componenttest/containers/ArtifactoryImageNameSubstitutor.java

```diff
@@ -30,21 +39,26 @@ public class ArtifactoryImageNameSubstitutor extends ImageNameSubstitutor {
 
     @Override
     public DockerImageName apply(DockerImageName original) {
-        // Priority 1: If we are using a programmatically built image, or a registry was explicit set, do not substitute
-        if (isSyntheticImage(original) || (original.getRegistry() != null && !original.getRegistry().isEmpty())) {
+        // Priority 1: If we are using a synthetic image do not substitute nor cache
+        if (isSyntheticImage(original)) {
```

Comment:
Could you explain why we don't want to cache in this case?

Reply:
A synthetic image is one that is built at runtime.  We discourage the use of synthetic images because of the performance impact. But we still check for them in our test infrastructure and handle them correctly. 
We wouldn't want to cache their names because by their nature they are never stored in GitHub/Artifactory. 
Their names are typically a random hash like: `localhost/testcontainers/jxbqdhfgsomphxr5:latest`